lang_js/analyze/graph_code_js.ml

```diff
@@ -179,11 +181,11 @@ let add_locals env vs =
      ) in
   { env with locals = locals @ env.locals } 
 
-let kind_of_expr v_kind e =
-  match e with
-  | Fun _ -> E.Function
-  | Class _ -> E.Class
-  | Obj _ -> E.Class
+let kind_of_expr_opt v_kind eopt =
+  match eopt with
+  | Some (Fun _) -> E.Function
+  | Some (Class _) -> E.Class
+  | Some (Obj _) -> E.Class
```

Comment:
I think you need a `None` case here? Unless we imagine "no expression" to be a global value.

Reply:
the "_" case will cover the None, and yes if you do var x;  without any value (eopt = None)
in this case, we consider this a global/constant (depending on vkind).

Anyway, this file is not used by semgrep, it's for codegraph.